shared/teams/new-team/container.js

```diff
@@ -17,31 +18,41 @@ const mapStateToProps = (state: TypedState) => ({
 })
 
 const mapDispatchToProps = (dispatch: Dispatch, {navigateUp, routePath}) => ({
-  _onCreateNewTeam: (teamname: string) => {
+  _onCreateNewTeam: (joinSubteam: boolean, teamname: string) => {
     const rootPath = routePath.take(1)
     const sourceSubPath = routePath.rest()
     const destSubPath = sourceSubPath.butLast()
-    dispatch(TeamsGen.createCreateNewTeam({teamname, rootPath, sourceSubPath, destSubPath}))
+    dispatch(TeamsGen.createCreateNewTeam({destSubPath, joinSubteam, rootPath, sourceSubPath, teamname}))
   },
   _onSetTeamCreationError: (error: string) => {
     dispatch(TeamsGen.createSetTeamCreationError({error}))
   },
   onBack: () => dispatch(navigateUp()),
 })
 
-const mergeProps = (stateProps, dispatchProps, ownProps) => ({
-  ...stateProps,
-  ...dispatchProps,
-  name: ownProps.routeProps.get('name'),
-})
+const mergeProps = (stateProps, dispatchProps, ownProps) => {
+  const name = ownProps.routeProps.get('name')
+  const baseTeam = name && baseTeamname(name)
+  const isSubteam = baseTeam && validTeamname(baseTeam)
+  return {
+    ...stateProps,
+    ...dispatchProps,
+    baseTeam,
+    isSubteam,
+    name,
+  }
+}
 
 export default compose(
   connect(mapStateToProps, mapDispatchToProps, mergeProps),
   withStateHandlers(({name}) => ({name: name || ''}), {
     onNameChange: () => (name: string) => ({name: name.toLowerCase()}),
   }),
+  withStateHandlers(({joinSubteam}) => ({joinSubteam: false}), {
```

Comment:
you can just merge this into the above. likely want to reduce all the nested hocs if we can